### PR TITLE
Isolated declarations expando functions

### PR DESCRIPTION
**Pull request by dragomirtitian**
Review after https://github.com/bloomberg/TypeScript/pull/118 is merged.

### CODE REVIEW

**Comment by h-joo** [comment]
I'm going to review this tomorrow, after #118  is merged.

**Comment by dragomirtitian** [comment]
@h-joo Thank you :)

File: src/compiler/transformers/declarations.ts [hunk 1]
```diff
@@ -1702,14 +1703,23 @@ export function transformDeclarations(context: TransformationContext) {
                     /*body*/ undefined,
                 ));
                 if (clean && resolver.isExpandoFunction(input) && shouldEmitFunctionProperties(input)) {
+                    const props = resolver.getPropertiesOfContainerFunction(input);
```
Comment (h-joo) [inline src/compiler/transformers/declarations.ts:1706]: Can you move this into the if statement? It doesn't seems to be used outside.

Or maybe `resolver.getPropertiesOfContainerFunction(input).forEach(...`

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDefaultExportWithStaticAssignment.d.ts.diff
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/declarationEmitDefaultExportWithStaticAssignment.d.ts.diff:1]: This change looks great.

File: src/compiler/transformers/declarations/emitBinder.ts [hunk 2]
```diff
@@ -369,171 +435,280 @@ export function bindSourceFileForDeclarationEmit(file: SourceFile) {
             else {
                 Debug.assertNever(d.name);
             }
+            function isExportedVariable(d: VariableDeclaration | ParameterDeclaration) {
+                if (!isVariableDeclaration(d)) return false;
```
Comment (h-joo) [inline src/compiler/transformers/declarations/emitBinder.ts]: Rather do `if(isParameterDeclaration(d)) return false` to be more explicit?

**Review by h-joo — APPROVED** [review]

File: src/compiler/transformers/declarations.ts (hunk 1 above)
Comment (dragomirtitian) [reply src/compiler/transformers/declarations.ts:1706]: `props` are used below as well

File: src/compiler/transformers/declarations/emitBinder.ts (hunk 2 above)
Comment (dragomirtitian) [reply src/compiler/transformers/declarations/emitBinder.ts]: Yeah. that sounds better. Changed

**Pull request merged**